skimage/transform/_warps.py

```diff
@@ -717,19 +717,34 @@ def _clip_warp_output(input_image, output_image, mode, cval, clip):
 
     """
     if clip:
-        min_val = input_image.min()
-        max_val = input_image.max()
-
-        preserve_cval = (mode == 'constant' and not
-                         (min_val <= cval <= max_val))
-
-        if preserve_cval:
-            cval_mask = output_image == cval
-
-        np.clip(output_image, min_val, max_val, out=output_image)
-
-        if preserve_cval:
-            output_image[cval_mask] = cval
```

Comment:
Just to make sure that I understand the way `clip` and `cval` interact together: if `cval` is outside the range of the input image and `clip=True`, what takes precedence? I think the old behavior tried to give `cval` precedence, although I think that `output_image == cval` never worked as intended. When I inspect the output image, the "outside area" seems to be close to but not precisely `cval`... So I wonder if `cval_mask` ever evaluated to true?